File: Data Structures & Algorithms/longest-common-prefix/submission-1.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        n = 10**18
        res = ""
        for s in strs:
            n = min(n, len(s))

        for i in range(n):
            temp = strs[0][i]
            for s in strs:
                if s[i] != temp:
                    return res
            res = res + temp
            
        return res

File: Data Structures & Algorithms/verifying-an-alien-dictionary/submission-1.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) 
    {
        int temp = 0;
        unordered_map<char, int> sequence;
        for(const char& ch : order)
            sequence[ch] = temp++;

        for(int i=0; i< words.size()-1; i++)
        {
            int size = min(words[i].size(), words[i+1].size());
            int j = 0;
            while(j < size)
            {
                if(words[i][j] != words[i+1][j])
                {
                    if(sequence[words[i][j]] > sequence[words[i+1][j]])
                        return false;
                    break;
                }
                j++;
            }
            if(j == size)
            {
                if(words[i].size() > words[i+1].size())
                    return false;
            }
        }
        return true;
    }
};